aten/src/ATen/Context.cpp

```diff
@@ -189,11 +189,16 @@ void Context::alertCuBLASConfigNotDeterministic() const {
 }
 
 bool Context::benchmarkCuDNN() const {
-  return benchmark_cudnn;
+  if (hasHIP()) {
```

Comment:
This must be returning false, which means for ROCm builds the `benchmark_cudnn` value is returned as false and forces MIOpen into immediate mode.  This is revealing a mem access fault in some kernel somewhere that MIOpen is using instead of its optimal benchmark-enabled kernel.

I'm assuming that `hasHIP()` is returning false because of aten/src/ATen/detail/HIPHooksInterface.h implements it to return false?

I know you don't want USE_ROCM, but if you hard-coded hasHIP() to return true it would do that for CUDA builds, as well.

How else besides USE_ROCM can we query at compile-time or run-time to distinguish if we are truly cuda or rocm-masquerading-as-cuda?

Reply:
I see, let me take a look. hasHIP should work...